plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -98,6 +105,139 @@ instance Pretty ChainIndexTxOut where
     pretty ScriptChainIndexTxOut {_ciTxOutAddress, _ciTxOutValue} =
                 hang 2 $ vsep ["-" <+> pretty _ciTxOutValue <+> "addressed to", pretty _ciTxOutAddress]
 
+-- TODO Move to cardano-api
+deriving instance Eq (C.EraInMode era C.CardanoMode)
+
+-- TODO Move to cardano-api
+instance FromJSON (C.EraInMode C.ByronEra C.CardanoMode) where
+  parseJSON "ByronEraInCardanoMode" = pure C.ByronEraInCardanoMode
+  parseJSON invalid =
+      prependFailure "parsing 'EraInMode ByronEra CardanoMode' failed, "
+                     (typeMismatch "ByronEraInCardanoMode" invalid)
+
+-- TODO Move to cardano-api
+instance FromJSON (C.EraInMode C.ShelleyEra C.CardanoMode) where
+  parseJSON "ShelleyEraInCardanoMode" = pure C.ShelleyEraInCardanoMode
+  parseJSON invalid =
+      prependFailure "parsing 'EraInMode ShelleyEra CardanoMode' failed, "
+                     (typeMismatch "ShelleyEraInCardanoMode" invalid)
```

Comment:
Better way to write those FromJSON instances which warns us when a new constructor is added in `C.EraInMode` ?

Reply:
Can't think of anything, might be worth asking in #haskell